ts/Core/Series/Series.ts

```diff
@@ -4792,7 +4791,7 @@ interface Series extends SeriesLike {
     directTouch: boolean;
     hcEvents?: Record<string, Array<U.EventWrapperObject<Series>>>;
     isCartesian: boolean;
-    kdAxisArray: Array<string>;
+    kdAxisArray: Array<'clientX' | 'plotY'>;
```

Comment:
I am not sure if we should limit the type of this property that strictly. For example, in GeoHeatmap interpolation I [build kd-tree based on lon, lats value](https://github.com/highcharts/highcharts/blob/4b54538e0e35fc614b7925eccb5f01c00c69a92e/ts/Series/GeoHeatmap/GeoHeatmapSeries.ts#L703). With that change it won't be clear enough for me (if we merge GeoHeatmap logic as it is).

Reply:
Should be fixed now but I might be safer to wait for your branch to be merged first so I'll probably wait with making this PR not a draft until then.